modules/ding_base/ding_base.module

```diff
@@ -1012,3 +1012,35 @@ function ding_base_taxonomy_autocomplete($vid, $tag) {
     drupal_json_output($items);
   }
 }
+
+/**
+ * Implements hook_cron().
+ */
+function ding_base_cron() {
```

Comment:
Please make it a one time action as an update hook instead.
Running this automatically canpotentially lead to unexpected, unrecorable results.
This is not common to have fields that dissapear by themselves to automate this. Instead, we need get a constant error message if this occurs and deal with this when needed.